pkg/util/docker/containers.go

```diff
@@ -182,6 +184,54 @@ func parseContainerHealth(status string) string {
 	return all[0][1]
 }
 
+// parseContainerNetworkAddresses converts docker ports
+// and network settings into a list of NetworkAddress
+func parseContainerNetworkAddresses(ports []types.Port, netSettings *types.SummaryNetworkSettings, container string) []containers.NetworkAddress {
+	addrList := []containers.NetworkAddress{}
+	tempAddrList := []containers.NetworkAddress{}
+	for _, port := range ports {
+		if isExposed(port) {
+			IP := net.ParseIP(port.IP)
+			if IP == nil {
+				log.Warnf("Unable to parse IP: %v for container: %s", port.IP, container)
+				continue
+			}
+			addrList = append(addrList, containers.NetworkAddress{
+				IP:       IP,                   // Host IP, since the port is exposed
+				Port:     int(port.PublicPort), // Exposed port
+				Protocol: port.Type,
+			})
+		} else {
+			// Port is not exposed, cache it and retieve IPs from network settings
+			tempAddrList = append(tempAddrList, containers.NetworkAddress{
+				Port:     int(port.PrivatePort),
+				Protocol: port.Type,
+			})
+		}
+	}
+	for _, network := range netSettings.Networks {
```

Comment:
are we sure that `netSettings` is never `nil` ?

Reply:
you are right it could be `nil` if we're using an older version of Docker, I'm fixing that